### PR TITLE
Fix stanley setStrategy method

### PR DESCRIPTION
**Pull request by pete-ipor**

### CODE REVIEW

**Review by alvaro-ipor — APPROVED** [review]

File: test/vault/unit/stanley/Stanley-SetStrategy.ts [hunk 1]
```diff
@@ -104,6 +105,42 @@ describe("Stanley -> StrategyChanged", () => {
...
+            await newStrategyAave.setShareToken(aDAI.address);
+
+            await newStrategyAave.setAsset(DAI.address);
+            const oldStrategyAddress = strategyAave.address;
+
+            const oldStrategyBalanceBefore = await strategyAave.balanceOf();
+            const newStrategyBalanceBefore = await newStrategyAave.balanceOf();
+
+            //when
+            await expect(stanley.setStrategyAave(newStrategyAave.address))
```
Comment (amydlarz) [inline test/vault/unit/stanley/Stanley-SetStrategy.ts:121]: Please extract stanley.setStrategyAave(newStrategyAave.address) to named constant. I don't know what is returning by setStrategyAave. In general, it is strange that the set is returning something.

File: test/vault/unit/stanley/Stanley-SetStrategy.ts [hunk 2]
```diff
@@ -104,6 +105,42 @@ describe("Stanley -> StrategyChanged", () => {
...
+                .withArgs(
+                    admin.getAddress,
+                    oldStrategyAddress,
+                    newStrategyAave.address,
+                    aDAI.address
+                );
+            //then
+            const newStrategyBalanceAfter = await newStrategyAave.balanceOf();
+            const oldStrategyBalanceAfter = await strategyAave.balanceOf();
+            expect(
```
Comment (amydlarz) [inline test/vault/unit/stanley/Stanley-SetStrategy.ts:133]: maybe 
`expect(newStrategyBalanceAfter).to.be.equal(ZERO);` ?

File: contracts/vault/Stanley.sol [hunk 3]
```diff
@@ -365,56 +365,63 @@ abstract contract Stanley is
...
-            );
         }
+    }
 
-        emit StrategyChanged(
-            _msgSender(),
-            oldStrategyAddr,
-            newStrategyAddr,
-            address(newShareToken)
+    function _revokeStrategyAllowance(address oldStrategyAddress) internal {
```
Comment (amydlarz) [inline contracts/vault/Stanley.sol:419]: this method can revoke not only oldStrategy so I think you can rename oldStrategyAddress to strategyAddress

File: test/vault/unit/stanley/Stanley-SetStrategy.ts (hunk 1 above)
Comment (pete-ipor) [reply test/vault/unit/stanley/Stanley-SetStrategy.ts:121]: This doesn't return anything, 
this is construction to catch events in tests with hardhat

**Review by amydlarz — APPROVED** [review]

**Pull request merged**